### PR TITLE
sg: Use internal migration utils

### PR DESCRIPTION
**Pull request by efritz**
This PR updates `sg migration {up|down}` to use the new migration utilities that will replace golang-migrate. See #28150 for a similar PR using the same migration infrastructure to power `cmd/migrator`.

### CODE REVIEW

**Review by eseliger — APPROVED** [review]

**Review by mrnugget — APPROVED** [review]
LGTM! But I'm wondering whether this shouldn't result in more dead code in `sg`? 🤔

File: dev/sg/sg_migration.go [hunk 1]
```diff
@@ -24,32 +28,11 @@ var (
...
-	migrationDownCommand          = &ffcli.Command{
-		Name:       "down",
-		ShortUsage: fmt.Sprintf("sg migration down [-db=%s] [-n=1]", db.DefaultDatabase.Name),
-		ShortHelp:  "Run down migration files",
-		FlagSet:    migrationDownFlagSet,
-		Exec:       migrationDownExec,
-		LongHelp:   constructMigrationSubcmdLongHelp(),
-	}
+	upCommand   = cliutil.Up("sg migration", runMigration, out)
+	downCommand = cliutil.Down("sg migration", runMigration, out)
```
Comment (mrnugget) [inline dev/sg/sg_migration.go:35]: Can you take a look at https://github.com/sourcegraph/sourcegraph/blob/d605622a978c1b383c480deda3c9a99ae9ca600f/doc/dev/background-information/sg/index.md#L175-L195 to see whether the flags still work with `--`?

Comment (efritz) [reply dev/sg/sg_migration.go:37]: It do!

**Comment by efritz** [comment]
@mrnugget There will be more once we simplify the assumptions, but the remaining code is all about adding/squashing migrations. There wasn't a lot shared there.

**Pull request merged**